src/base/bittorrent/session.cpp

```diff
@@ -371,6 +371,9 @@ Session::Session(QObject *parent)
     , m_useProxy(false)
     , m_recentErroredTorrentsTimer(new QTimer(this))
 {
+    if (useRandomPort() || (port() < 0))
+        m_port = Utils::Random::rand(1024, 65535);
```

Comment:
It would probably be better to assign a port in the dynamic port range, to avoid conflicts with other well-known assignments.
https://www.iana.org/assignments/service-names-port-numbers/service-names-port-numbers.xhtml

```suggestion
        m_port = Utils::Random::rand(49152, 65535);
```